src/lib389/lib389/utils.py

```diff
@@ -2013,3 +2013,14 @@ def check_plugin_strings(plugin_name, tested_strings):
                 return { astring:(mm.find(astring.encode()) >=0) for astring in tested_strings }
     return { astring:None for astring in tested_strings }
 
+
+def get_timeout_scale():
+    """
+    Get the timeout scale factor
+    :return float:
+    """
+    scale_factor = 1.0
+    if Paths().asan_enabled:
+        scale_factor = 2.0
+    return float(os.getenv('DS_TIMEOUT_SCALE', default=scale_factor))
```

Comment:
Might worth to validate the env var here. It'll raise a bit cryptic (for user) message otherwise (for example, if `DS_TIMEOUT_SCALE=testvar`):

```
ValueError: could not convert string to float: 'testvar'
```

Reply:
What behaviour do you want in case the validation fails? Set the timeout scale to 1.0 or log some error? This env var is not meant to be used by the end user, it's for us to deal with occasional slowness of the test infra.

Reply:
I think an error that mentions the env var will work the best. It will help us to see exactly where is issue (and it'll be future-proof, in case anyone else will use the internal feature for some reason)

Reply:
IMHO we should check that it is a float and its value is >= 1/timeout (and just tell that a float >= 0.2 is expected 
Even if it is only for tester, a typo is so quick to do and a clear message will avoid us spending too much time to understand whats is happening!

Reply:
I've added an error with a fallback to the default value in case it's an invalid value:
```
ERROR:lib389.utils:DS_TIMEOUT_SCALE should be a valid float. Using default value: 2.0
```